logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/auth/autoVersioningAuth/AutoVersionAuthScopeUseCase.kt

```diff
@@ -11,11 +11,11 @@ class AutoVersionAuthScopeUseCase(
     private val serverLinks: ServerConfig.Links,
     private val coreLogic: CoreLogicCommon
 ) {
-    suspend operator fun invoke(): Result =
+    suspend operator fun invoke(proxyCredentials: (() -> Pair<String, String>)?): Result =
```

Comment:
I like the fact that this doesn't have a default argument.

This way you the caller needs to explicitly opt out of providing credentials.

Maybe it would be cleaner / more explicit if we had a:

```kotlin

sealed interface ProxyCredentials {

    object None: ProxyCredentials

    class UsernameAndPassword(val username: String, val password: String): ProxyCredentials
}

```

This way the caller needs to pass `ProxyCredentials.None` or `ProxyCredentials.UsernameAndPassword(username = "user@example.org", password = "123456")`

Reducing a bit the risk of calling `Pair(password, email)` instead of `Pair(email, password)` by accident.